crates/red_knot/src/types.rs

```diff
@@ -46,6 +51,45 @@ impl Type {
     pub const fn is_unknown(&self) -> bool {
         matches!(self, Type::Unknown)
     }
+
+    // NOTE: naming? `get_member_type`
+    pub fn get_member(&self, db: &dyn SemanticDb, name: &Name) -> QueryResult<Type> {
+        // NOTE: maybe make this a trait that each TypeID implements?
+        match self {
+            Type::Any => todo!("attribute lookup on Any type"),
+            Type::Never => todo!("attribute lookup on Never type"),
+            Type::Unknown => todo!("attribute lookup on Unknown type"),
+            Type::Unbound => todo!("attribute lookup on Unbound type"),
+            Type::Function(_) => todo!("attribute lookup on Function type"),
+            Type::Module(module_id) => module_id
+                .get_member(db, name)
+                // FIXME: don't want to unwrap here; need to add a QueryError ctor for the case
```

Comment:
I think it would be okay to change the signature of `get_member` to return `QueryResult<Option<Type>>`.  I don't think we should panic because it's likely that a lint rule uses this API to test if a class has a specific member.

Reply:
Yes, I agree that this method should return `QueryResult<Option<Type>>`. "No such member" and "query error (e.g. cancelled)" are distinct results that should be distinctly represented (and the "no such member" result is not an error.)

Reply:
Yes, this is obvious in hindsight. :sweat_smile: ty